File: 1509-minimum-difference-between-largest-and-smallest-value-in-three-moves/1509-minimum-difference-between-largest-and-smallest-value-in-three-moves.java
class Solution {
    public int minDifference(int[] nums) {
        Arrays.sort(nums);
        int n = nums.length, ans = Integer.MAX_VALUE;
        int i = 0, j = n-4;
        while(j < n && j >= 0) {
            ans = Math.min(ans, nums[j++] - nums[i++]);
        }
        return ans == Integer.MAX_VALUE ? 0 : ans;
    }
}

File: 1509-minimum-difference-between-largest-and-smallest-value-in-three-moves/1509-minimum-difference-between-largest-and-smallest-value-in-three-moves.cpp
class Solution {
public:
    int minDifference(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int n = nums.size(), ans = INT_MAX;
        int i = 0, j = n-4;
        while(j < n && j >= 0) {
            ans = min(ans, nums[j++] - nums[i++]);
        }
        return ans == INT_MAX ? 0 : ans;
    }
};